webhook/resourcesemantics/conversion/controller.go

```diff
@@ -32,50 +30,9 @@ import (
 	pkgreconciler "knative.dev/pkg/reconciler"
 	"knative.dev/pkg/system"
 	"knative.dev/pkg/webhook"
+	"knative.dev/pkg/webhook/resourcesemantics/common"
 )
 
-// ConvertibleObject defines the functionality our API types
-// are required to implement in order to be convertible from
-// one version to another
-//
-// Optionally if the object implements apis.Defaultable the
-// ConversionController will apply defaults before returning
-// the response
-type ConvertibleObject interface {
-	// ConvertTo(ctx, to)
-	// ConvertFrom(ctx, from)
-	apis.Convertible
-
-	// DeepCopyObject()
-	// GetObjectKind() => SetGroupVersionKind(gvk)
-	runtime.Object
-}
-
-// GroupKindConversion specifies how a specific Kind for a given
-// group should be converted
-type GroupKindConversion struct {
-	// DefinitionName specifies the CustomResourceDefinition that should
-	// be reconciled with by the controller.
-	//
-	// The conversion webhook configuration will be updated
-	// when the CA bundle changes
-	DefinitionName string
-
-	// HubVersion specifies which version of the CustomResource supports
-	// conversions to and from all types
-	//
-	// It is expected that the Zygotes map contains an entry for the
-	// specified HubVersion
-	HubVersion string
-
-	// Zygotes contains a map of version strings (ie. v1, v2) to empty
-	// ConvertibleObject objects
-	//
-	// During a conversion request these zygotes will be deep copied
-	// and manipulated using the apis.Convertible interface
-	Zygotes map[string]ConvertibleObject
-}
```

Comment:
Deleting these types is unfortunately a breaking change

Reply:
Yes, it isn't nice. We'll need to keep this type untouched.